src/codeu/chat/client/core/Controller.java

```diff
@@ -34,11 +35,17 @@
   private final static Logger.Log LOG = Logger.newLog(Controller.class);
 
   private final ConnectionSource source;
+  private User user;
 
   public Controller(ConnectionSource source) {
     this.source = source;
   }
 
+  public Controller(Controller controller, User user) {
+    this(controller.source);
```

Comment:
ah, ok, i see it now.

consider adding a method "public Controller withUser(User user) { return new Controller(this, user); }" instead of a new constructor.